### PR TITLE
Feature/uds read dtc

### PR DESCRIPTION
**Pull request by RubenNatanael**
Implementation of the 0x19  UDS Service, Read DTC information:

- Sub-funct 0x01 Read number of dtc  by staus mask.
- Sub-funct 0x02 Read dtc by staus mask.
- Unit test(coverage 100%)
- Add to GenerateFrames method for response to sub-fun 0x02

### CODE REVIEW

File: src/uds/read_dtc_information/include/read_dtc_information.h [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/**
```
Comment (Mihnea9901) [inline src/uds/read_dtc_information/include/read_dtc_information.h:1]: Rename the files like this: ReadDtcInformation
And also please add service call in handleFrames

**Review by TheoRandstad — COMMENTED** [review]
For me its not clear where you treat the NRC scenarios(Logic + sending Nrc frame, after NRC merge we just  replace the sendFrame). Also, its not clear where do you send the response(You send back with the same id)

File: src/uds/read_dtc_information/include/read_dtc_information.h (hunk 1 above)
Comment (RubenNatanael) [reply src/uds/read_dtc_information/include/read_dtc_information.h]: Done

**Comment by RubenNatanael** [comment]
Implemented NRC, id, resolved conflicts in order to marge

**Review by Mihnea9901 — APPROVED** [review]

**Review by TheoRandstad — APPROVED** [review]
Fine from me too.

**Pull request merged**